### PR TITLE
Implemented Compare component

### PR DESCRIPTION
**Pull request by sglogovic**
Added compare component to quick-react-ts.

Component displays left and right grid with shared scroll based on row and column data.
Component props :

- sourceRows - Collection of row data for left grid
- targetRows - Collection of row data for right grid
- columns - Collection of column data for both grids
- differences - Collection of rowIndexes, columnIndexes and differenceTypes 

NOTE : 
This component does not compare data form source and target row, differences property recieves what is different and what is not.

image

### CODE REVIEW

File: src/components/Compare/CompareComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                                    <div className={'grid-component-container'} style={{height : height - 50}}>
+                                        <div className={'compare-single-grid-wrapper'}>
+                                            <GridHeader
+                                                allColumns={this.columns}
+                                                headerColumns={this.columns}
+                                                columnWidths={this.state.columnWidths}
+                                                width={this.gridComponentWidth(width)}
+                                                scrollLeft={scrollLeft}
+                                                className={'grid-component-header'}
+                                                groupBy={[]}
```
Comment (hbagarich) [inline src/components/Compare/CompareComponent.tsx]: this cause rerender always. Maybe ommit this prop or use const?

Comment (sglogovic) [reply src/components/Compare/CompareComponent.tsx]: Made groupBy optional in QuickGridHeader as suggested.

**Comment by sglogovic** [comment]
Made groupBy optional in QuickGridHeader as suggested.

**Comment by sglogovic** [comment]
Added defaultProperty value for groupBy to avoid rerendering.

**Pull request merged**